scalafix-testkit/src/main/scala/scalafix/testkit/DiffTest.scala

```diff
@@ -25,49 +25,61 @@ object DiffTest {
   private def stripPrefix(str: String) = PrefixRegex.replaceFirstIn(str, "")
 
   def fromSemanticdbIndex(index: SemanticdbIndex): Seq[DiffTest] = {
-    index.documents.map { document =>
+    index.documents.flatMap { document =>
       val input = document.input
-      val (label, code) =
-        document.input match {
-          case Input.VirtualFile(label, code) => (label, code)
-          case file: Input.File => {
-            val relative = index.sourcepath.shallow
-              .find(sourceroot => file.path.toNIO.startsWith(sourceroot.toNIO))
-              .map(root => file.path.toRelative(root))
-              .getOrElse(throw new Exception(
-                s"${file.path} is not relative to any sourcroot"))
+      document.input match {
+        case file: Input.File => {
+          val relative = index.sourcepath.shallow
+            .find(sourceroot => file.path.toNIO.startsWith(sourceroot.toNIO))
+            .map(root => file.path.toRelative(root))
+            .getOrElse(throw new Exception(
+              s"${file.path} is not relative to any sourcroot"))
 
-            (relative.toString(), file.text)
+          val label = relative.toString()
+          val code = file.text
+
+          val relpath = RelativePath(label)
+
+          val config: () => (Rule, ScalafixConfig) = { () =>
+            input.tokenize.get
+              .collectFirst {
+                case Token.Comment(comment) =>
+                  val decoder =
+                    ScalafixReflect.fromLazySemanticdbIndex(
+                      LazySemanticdbIndex(_ => Some(index)))
+                  ScalafixConfig
+                    .fromInput(
+                      Input.VirtualFile(label, stripPrefix(comment)),
+                      LazySemanticdbIndex(_ => Some(index)))(decoder)
+                    .get
+              }
+              .getOrElse(throw new TestFailedException(
+                s"Missing scalafix configuration inside comment at top of file $relpath",
+                0))
           }
-          case _ => throw new Exception("unexpected")
+          Seq(
+            DiffTest(
+              filename = relpath,
+              original = input,
+              document = document,
+              config = config,
+              isSkip = code.contains("SKIP"),
+              isOnly = code.contains("ONLY")
+            ))
         }
+        case Input.VirtualFile(label, _) =>
+          println(s"ignoring: $label")
+          // I don't know why, we get those as VirtualFile:
+          // ignoring: scalafix-tests/shared/src/main/scala/test/SymbolMatcherTest.scala
+          // ignoring: scalafix-tests/shared/src/main/scala/test/DenotationOpsTest.scala
+          // ignoring: scalafix-tests/shared/src/main/scala/test/SemanticdbIndexTest.scala
+          // ignoring: scala/sbtfix/Sbt1.scala
```

Comment:
@olafurpg Do you know why this has changed? I think this is since I bumped the scalameta version.

Reply:
this was because of the slim mode!